File: 0476 - Number Complement/NumberComplement.cpp
#include <math.h>

class Solution {
public:
    int findComplement(int num) {
        int ans = 0;
        int exp = 0;
        while (num > 0) {
            if (num % 2 == 0)
                ans += (int)pow(2, exp);
            exp++;
            num /= 2;
        }
        return ans;
    }
};

File: 0476 - Number Complement/NumberComplement.py
class Solution:
    def findComplement(self, num: int) -> int:
        exp = 0
        ans = 0
        while num > 0:
            current = round(num % 2)
            if current == 0:
                ans += pow(2, exp)
            exp += 1
            num = math.floor(num / 2)
        
        return ans